### PR TITLE
Partitioning UI

### PR DESCRIPTION
**Pull request by SvetaMemphis**

### CODE REVIEW

File: ui_src/src/domain/stationOverview/stationObservabilty/components/messageDetails/index.js [hunk 1]
```diff
@@ -45,20 +45,22 @@ const MessageDetails = ({ isDls, isFailedSchemaMessage = false }) => {
     }, [messageDetails]);
 
     useEffect(() => {
-        if (stationState?.selectedRowId && !loadMessageData) {
-            getMessageDetails(stationState?.selectedRowId);
+        if ((isDls && stationState?.selectedRowId && stationState?.selectedRowPartition && !loadMessageData) || (stationState?.selectedRowId && !loadMessageData)) {
```
Comment (avrhamNeeman) [inline ui_src/src/domain/stationOverview/stationObservabilty/components/messageDetails/index.js:48]: Try to change the names of the fields.

File: ui_src/src/domain/stationOverview/stationObservabilty/components/purgeStationModal/index.js [hunk 2]
```diff
@@ -46,20 +46,16 @@ const PurgeStationModal = ({ title, desc, cancel, stationName, msgsDisabled = fa
...
-            // stationDispatch({ type: 'SET_SOCKET_DATA', payload: data });
-            // setLoader(false);
-            // cancel();
+            let data = stationState?.stationSocketData;
+            if (purgeDataPayload['purge_station']) {
+                data['total_messages'] = 0;
+                data['messages'] = [];
+            }
+            if (purgeDataPayload['purge_dls']) data['total_dls_messages'] = 0;
+            stationDispatch({ type: 'SET_SOCKET_DATA', payload: data });
```
Comment (avrhamNeeman) [inline ui_src/src/domain/stationOverview/stationObservabilty/components/purgeStationModal/index.js:58]: remove it

File: ui_src/src/components/createStationForm/index.js [hunk 3]
```diff
@@ -295,33 +295,63 @@ const CreateStationForm = ({ createStationFormRef, getStartedStateRef, finishUpd
...
                         <TitleComponent
-                            headerTitle="Replicas"
+                            headerTitle="Partitions"
                             typeTitle="sub-header"
-                            headerDescription="Amount of mirrors per message."
-                            learnMore={true}
-                            link="https://docs.memphis.dev/memphis/memphis/concepts/station#replicas-mirroring"
+                            headerDescription="Amount of partitions per station."
+                            learnMore={false}
+                            // link="https://docs.memphis.dev/memphis/memphis/concepts/station#replicas-mirroring"
```
[the remark this answers is not in the document]
Comment (SvetaMemphis) [reply ui_src/src/components/createStationForm/index.js:336]: Done

File: ui_src/src/components/createStationForm/style.scss [hunk 4]
```diff
@@ -140,30 +145,32 @@
...
+                margin: 0;
             }
         }
         .separator {
-            margin: 3px 5px 0px 5px;
+            margin: 8px 5px 0px 5px;
             font-size: 14px;
             font-family: 'InterBold';
-            line-height: 23px;
+            // line-height: 20px;
```
[the remark this answers is not in the document]
Comment (SvetaMemphis) [reply ui_src/src/components/createStationForm/style.scss:160]: Done

File: ui_src/src/components/createStationForm/style.scss [hunk 5]
```diff
@@ -140,30 +145,32 @@
...
             color: var(--gray);
         }
     }
 
     .ant-input-number {
-        width: 60px;
-        height: 38px;
+        // width: 60px;
```
[the remark this answers is not in the document]
Comment (SvetaMemphis) [reply ui_src/src/components/createStationForm/style.scss:166]: done

File: ui_src/src/components/partitionsFilter/index.js [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+                <div className="filter-partitions-title">
+                    <p>Filter</p>
+                    <Divider />
+                </div>
+                <div className="filter-partitions-container">{getItems()}</div>
+            </div>
+        );
+    };
+
+    <div>{partitions_number}</div>;
```
[the remark this answers is not in the document]
Comment (SvetaMemphis) [reply ui_src/src/components/partitionsFilter/index.js:61]: removed

File: ui_src/src/components/partitionsFilter/style.scss [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+    color: #2e2c34;
+    p {
+        margin-bottom: 0;
+        padding: 10px 0 10px 20px;
+    }
+    .ant-divider {
+        margin: 0;
+    }
+}
+.ant-btn {
```
[the remark this answers is not in the document]
Comment (SvetaMemphis) [reply ui_src/src/components/partitionsFilter/style.scss:24]: removed

File: ui_src/src/components/partitionsFilter/style.scss [hunk 8]
```diff
@@ -0,0 +1,76 @@
+.ant-popover {
+    padding-top: 0;
+}
+
+.ant-popover-inner {
+    border-radius: 8px;
+}
```
[the remark this answers is not in the document]
Comment (SvetaMemphis) [reply ui_src/src/components/partitionsFilter/style.scss:7]: removed

File: ui_src/src/domain/overview/failedStations/index.js [hunk 9]
```diff
@@ -66,6 +67,12 @@ const FailedStations = ({ createStationTrigger }) => {
                                             <OverflowTip className="station-creation" text={parsingDate(station.created_at)}>
                                                 {parsingDate(station.created_at)}
                                             </OverflowTip>
+                                            <OverflowTip
+                                                className="station-details total"
+                                                text={station?.partitions_list ? station?.partitions_list?.length?.toLocaleString() : 0}
```
[the remark this answers is not in the document]
Comment (SvetaMemphis) [reply ui_src/src/domain/overview/failedStations/index.js:72]: done

File: ui_src/src/domain/overview/failedStations/style.scss [hunk 10]
```diff
@@ -38,11 +38,12 @@
             padding: 15px 0 15px 15px;
             width: 100%;
             display: grid;
-            grid-template-columns: 25% 20% 20% 13% 15%;
-            gap:5px;
+            // grid-template-columns: 25% 20% 20% 13% 15%;
```
[the remark this answers is not in the document]
Comment (SvetaMemphis) [reply ui_src/src/domain/overview/failedStations/style.scss:41]: done

File: ui_src/src/domain/overview/failedStations/style.scss [hunk 11]
```diff
@@ -64,8 +65,9 @@
             padding: 0 0 5px 15px;
             .stations-row {
                 display: grid;
-                grid-template-columns: 25% 20% 20% 13% 15%;
-                gap:5px;
+                // grid-template-columns: 25% 20% 20% 13% 15%;
```
[the remark this answers is not in the document]
Comment (SvetaMemphis) [reply ui_src/src/domain/overview/failedStations/style.scss:68]: done

File: ui_src/src/domain/stationOverview/index.js [hunk 12]
```diff
@@ -77,10 +81,14 @@ const StationOverview = () => {
...
         }
     };
+    useEffect(() => {
+        getStationDetails();
+    }, [stationState?.stationPartition || stationState?.stationSocketData?.total_messages || stationState?.stationSocketData?.total_dls_messages]);
 
     useEffect(() => {
         setisLoading(true);
         dispatch({ type: 'SET_ROUTE', payload: 'stations' });
+        stationDispatch({ type: 'SET_STATION_PARTITION', payload: -1 });
```
[the remark this answers is not in the document]
Comment (SvetaMemphis) [reply ui_src/src/domain/stationOverview/index.js:91]: removed

File: ui_src/src/domain/stationOverview/stationObservabilty/components/messageDetails/index.js (hunk 1 above)
Comment (SvetaMemphis) [reply ui_src/src/domain/stationOverview/stationObservabilty/components/messageDetails/index.js:48]: done

File: ui_src/src/domain/stationOverview/stationObservabilty/components/purgeStationModal/index.js (hunk 2 above)
Comment (SvetaMemphis) [reply ui_src/src/domain/stationOverview/stationObservabilty/components/purgeStationModal/index.js:58]: Moved to comment

**Comment by SvetaMemphis** [comment]
@avrhamNeeman All comments addressed

File: ui_src/src/domain/stationOverview/stationObservabilty/components/purgeStationModal/index.js [hunk 13]
```diff
@@ -74,7 +70,7 @@ const PurgeStationModal = ({ title, desc, cancel, stationName, msgsDisabled = fa
             setLoader(false);
             setPurgeData({});
         }
-    }, [stationState?.stationSocketData]);
+    }, [stationState?.stationSocketData?.total_messages || stationState?.stationSocketData?.total_dls_messages]);
```
[the remark this answers is not in the document]
Comment (SvetaMemphis) [reply ui_src/src/domain/stationOverview/stationObservabilty/components/purgeStationModal/index.js]: done

**Pull request merged**